### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by svelupillai**
Refactored some large classes, based off feedback from last release and code added in new release.

### CODE REVIEW

**Comment by svelupillai** [comment]
FavouritesOverviewTest passes:

image

**Comment by svelupillai** [comment]
FavouritesTest passes:

image

**Comment by svelupillai** [comment]
HintTest passes:

image

**Comment by svelupillai** [comment]
QuizHistoryTest Passes: 

image

**Comment by svelupillai** [comment]
ShuffleTest passes:

image

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        History result = HistoryHelper.addToHistory(score, count, hist, "/test", historyDao);
+        hist = newDbHelper.getHistoryForDB("/test");
+        assertEquals(10, hist.size());
+        assertEquals(50.0, result.getMark());
+
+    }
+
+    @SmallTest
+    @Test
+    public void testAverage(){
```
Comment (coreylevine) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java]: Add space between ){

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        for (History history: hist){
+            histories.add(history);
+        }
+        double average = HistoryHelper.computeAverage(histories);
+        assertEquals(59.5, average);
+    }
+
+    @SmallTest
+    @Test
+    public void testDate(){
```
Comment (coreylevine) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java]: Add space between ){

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    @SmallTest
+    @Test
+    public void testInsertFunction() {
+        int score = 50;
+        List<History> hist = newDbHelper.getHistoryForDB("/test");
+        int count = hist.size();
+        History result = HistoryHelper.addToHistory(score, count, hist, "/test", historyDao);
+        hist = newDbHelper.getHistoryForDB("/test");
+        assertEquals(10, hist.size());
+        assertEquals(50.0, result.getMark());
```
Comment (coreylevine) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java]: What is this assert checking?

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    @SmallTest
+    @Test
+    public void testDate(){
+        History history1 = new History("/test", 100, 15346);
+        String expected = "12/31/69 8:00 PM";
+        String est = HistoryHelper.estDate(history1.getTimeStamp());
+        assertEquals(expected, est);
+
```
Comment (coreylevine) [inline app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java]: Remove extra lines

**Comment by j-suguma** [comment]
There is a failing test:
image

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizActivity.java [hunk 5]
```diff
@@ -489,6 +485,11 @@ private void showCompleteAllDialog() {
 
     /* Called when all new cards are completed. */
     private void showCompleteNewDialog(int correct) {
+        if(mediaPlayer !=null){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizActivity.java]: Fix spacing --> if (....) {

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizActivity.java [hunk 6]
```diff
@@ -489,6 +485,11 @@ private void showCompleteAllDialog() {
...
+            if(mediaPlayer.isPlaying()){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/QuizActivity.java]: fix spacing --> if (....) {

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java [hunk 7]
```diff
@@ -456,33 +411,93 @@ public boolean onLongClick(View v) {
...
+                toggleSelection(position);
+            }
+        }
+
+        private void setMoreButton(RecentItem currentItem) {
+            String s = currentItem.dbPath;
+            String[] split = s.split(".db");
+            String nameWithoutDB = split[0];
+
+            if(nameWithoutDB.endsWith("_MC")) {
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java]: add space after if

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java [hunk 8]
```diff
@@ -456,33 +411,93 @@ public boolean onLongClick(View v) {
...
+            String nameWithoutDB = split[0];
+            if (getSelectedItemCount() == 0 && !nameWithoutDB.endsWith("_MC")) {
+                Intent myIntent = new Intent();
+                myIntent.setClass(context, StudyActivity.class);
+                String dbPath = currentItem.dbPath;
+                myIntent.putExtra(StudyActivity.EXTRA_DBPATH, dbPath);
+                myIntent.putExtra(SHUFFLE_CARDS, shuffle);
+                recentListUtil.addToRecentList(dbPath);
+                context.startActivity(myIntent);
+            } else if (getSelectedItemCount() == 0 && nameWithoutDB.endsWith("_MC"))  {
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/RecentListFragment.java]: remove extra space between ) and {

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/HistoryHelper.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        Date date = new Date();
+        Long timeStamp = date.getTime();
+        History result = new History();
+        result.setdbPath(dbPath);
+        result.setMark(score);
+        result.setTimeStamp(timeStamp);
+        historyDao.insertHistory(result);
+
+        return result;
+
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/HistoryHelper.java]: Remove extra space

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/HistoryHelper.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public static Double computeAverage (ArrayList<History> histories){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/HistoryHelper.java]: Add space between ){

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Created by SnegaT on 2018-04-12.
+ */
+
+public class SettingHelper {
+
+    public SettingHelper(){}
+
+    public static List setColors(Setting setting){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java]: Add space between ){

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+        colors.add(setting.getAnswerTextColor());
+        colors.add(setting.getQuestionBackgroundColor());
+        colors.add(setting.getHintBackgroundColor());
+        colors.add(setting.getAnswerBackgroundColor());
+        colors.add(setting.getSeparatorColor());
+
+        return colors;
+    }
+
+    public static Setting setValues(Setting setting, List<Integer> colors){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java]: Add space between ){

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        setting.setAnswerTextColor(colors.get(2));
+        setting.setQuestionBackgroundColor(colors.get(3));
+        setting.setHintBackgroundColor(colors.get(4));
+        setting.setAnswerBackgroundColor(colors.get(5));
+        setting.setSeparatorColor(colors.get(6));
+
+        return setting;
+    }
+
+    public static Bundle setQuestionBundle(Bundle b){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java]: Add space between ){

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+        b.putString(FileBrowserFragment.EXTRA_FILE_EXTENSIONS, ".ttf");
+        b.putString(FileBrowserFragment.EXTRA_DEFAULT_ROOT, AMEnv.DEFAULT_ROOT_PATH);
+        b.putBoolean(FileBrowserFragment.EXTRA_DISMISS_ON_SELECT, true);
+
+        return b;
+    }
+
+    public static Bundle setAnswerBundle(Bundle b){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java]: Add space between ){

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+        b.putString(FileBrowserFragment.EXTRA_FILE_EXTENSIONS, ".ttf");
+        b.putString(FileBrowserFragment.EXTRA_DEFAULT_ROOT, AMEnv.DEFAULT_ROOT_PATH);
+        b.putBoolean(FileBrowserFragment.EXTRA_DISMISS_ON_SELECT, true);
+
+        return b;
+    }
+
+    public static ProgressDialog setProgressDialog(ProgressDialog progressDialog, String title, String message){
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/helper/SettingHelper.java]: Add space between ){

**Review by coreylevine — CHANGES_REQUESTED** [review]
The refactoring looks really good! Make the code a lot more readable. Just a few ASOP things to fix.

**Comment by rekandiah** [comment]
Test that checks the last answer in a multiple choice card is selected when the user does not specify an answer.

Test to check a multiple choice card is edit and saved.

Test to check if user can play a multiple choice deck.



ALL PASSED

**Comment by j-suguma** [comment]
CI build refactor
image

tests:
image

static analysis:
There are no critical/high level warnings in the files that were changed so no changes need to be made
image

File: app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java (hunk 3 above)
Comment (svelupillai) [reply app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java]: checks that its still 10 entries even after the 11th quiz and that the right object is being inserted.

**Comment by svelupillai** [comment]
HintToggleCheckBoxTest passes :

image

**Review by coreylevine — APPROVED** [review]
Looks good!

**Review by rekandiah — APPROVED** [review]
I checked over the stuff you guys refactored and everything functions well! nice job :)

**Pull request merged**